src/trio/_tests/test_testing_raisesgroup.py

```diff
@@ -290,6 +490,349 @@ def check_message(
+            "      The following raised exceptions did not find a match\n"
+            "        TypeError('cccccccccccccccccccccccccccccccccccccccc'):\n"
+            "          RaisesGroup(ValueError): 'TypeError' is not an exception group\n"
+            "        TypeError('dddddddddddddddddddddddddddddddddddddddd'):\n"
+            "          RaisesGroup(ValueError): 'TypeError' is not an exception group\n"
+            "    RaisesGroup(Matcher(TypeError, match='foo')): \n"
+            "      The following expected exceptions did not find a match:\n"
+            "        Matcher(TypeError, match='foo')\n"
+            "      The following raised exceptions did not find a match\n"
+            "        TypeError('cccccccccccccccccccccccccccccccccccccccc'):\n"
+            "          Matcher(TypeError, match='foo'): Regex pattern 'foo' did not match 'cccccccccccccccccccccccccccccccccccccccc'\n"
+            "        TypeError('dddddddddddddddddddddddddddddddddddddddd'):\n"
+            "          Matcher(TypeError, match='foo'): Regex pattern 'foo' did not match 'dddddddddddddddddddddddddddddddddddddddd'\n"
+            "    RaisesGroup(TypeError, ValueError): \n"
+            "      1 matched exception. \n"
+            "      The following expected exceptions did not find a match:\n"
+            "        'ValueError'\n"
+            "      The following raised exceptions did not find a match\n"
+            "        TypeError('dddddddddddddddddddddddddddddddddddddddd'):\n"
+            "          It matches 'TypeError' which was paired with TypeError('cccccccccccccccccccccccccccccccccccccccc')\n"
+            "          'TypeError' is not of type 'ValueError'",
+            add_prefix=False,  # to see the full structure
+        ),
+        RaisesGroup(
+            RaisesGroup(ValueError),
+            RaisesGroup(RaisesGroup(ValueError)),
+            RaisesGroup(Matcher(TypeError, match="foo")),
+            RaisesGroup(TypeError, ValueError),
+        ),
+    ):
+        raise ExceptionGroup(
+            "",
+            [
+                TypeError("aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"),
+                ExceptionGroup(
+                    "Exceptions from Trio nursery",
+                    [TypeError("bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb")],
+                ),
+                ExceptionGroup(
+                    "Exceptions from Trio nursery",
+                    [
+                        TypeError("cccccccccccccccccccccccccccccccccccccccc"),
+                        TypeError("dddddddddddddddddddddddddddddddddddddddd"),
+                    ],
+                ),
+            ],
+        )
+    with (
+        pytest.raises(
+            AssertionError,
+            match=(
+                r"^Raised exception group did not match: \n"
+                r"The following expected exceptions did not find a match:\n"
+                r"  Matcher\(check=<function test_assert_message_nested.<locals>.<lambda> at .*>\)\n"
```

Comment:
ah... if we do the monkeypatch from Trio's hypothesis plugin as I suggest above, we'll then also want to use the maybe-monkeypatched `_check_repr` function to construct the expected string; and maybe also assert that that string is one of two known possibilities.  Which is kinda ugly, but imo showing the lambda source here is nice enough to be worth it.

Reply:
oh.. I just realized if `hypothesis` start always monkeypatching it as soon as it's installed then the test suite would fail on install as well. Or possible even in some scenarios of hypothesis-pytest-plugin thingie thingies.

I'll split out a test case showing how ugly the lambda repr is and make it skip if hypothesis is in `sys.modules`, and convert the rest to use `_check_repr`, to avoid future headaches.